### PR TITLE
Caerbannog

### PR DESCRIPTION
**Pull request by dylanmcreynolds**
Adds a working ingest, ingesting tiff sequences. Adds publishing containers to ghcr.io.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -20,18 +20,20 @@ classifiers = [
 
 dependencies = [
    "pika",
-   "tiled[client]"
-
+   "tiled[server]==0.1.0a114",
```
Comment (Wiebke) [inline pyproject.toml:23]: Just putting a reminder for us here to also update the version in mlex_tomo_framework.

File: src/ingest.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    else:
+        from pprint import pprint
+        import os
+        pprint(os.environ)
+        asyncio.run(
+            process_file(
+                # "/tiled_storage/beamlines/8.3.2/recons/rec20240207_120550_test_no_xrays_n257",
+                "/tiled_storage/beamlines/8.3.2/recons/rec20240207_120829_test_no_xrays_n1313",
+                path_prefix="/beamlines/8.3.2/recons/"
+            )
```
Comment (Wiebke) [inline src/ingest.py:87]: This is a bit of an unrealistic use case (and I doubt we will encounter it for the current project), but I wanted to document it regardless, as the test data for the segmentation application is currently downloaded this way.

I noticed that in this case (where `file_path` contains more components than a path contained in the readable storage list of the matching tree within the config and `path_prefix`), we may be loosing some identifying information. Specifically this is the case where the tiff-sequence within the given folder contains only numbers.

In that case on the naming of the tiff-sequences, we may consider adding the folder to the `path_prefix`. 

As an example, I have the following folder structure on filesystem:
```
├── tiled_storage
│   ├── reconstruction
│   │   └── check_handedness
│   │       ├── 0.tiff
│   │       └── 1.tiff
│   └── NXtomoproc.hdf5
└── ...
```

and ingest with:

```python
asyncio.run(
            process_file(
                "/tiled_storage/NXtomoproc.hdf5",
                path_prefix="/",
            )
        )

        asyncio.run(
            process_file(
                "/tiled_storage/reconstruction/check_handedness",
                path_prefix="/reconstruction",
            )
        )
```

which leads to container structure within Tiled as:

```
├── reconstruction
│   └── _unnamed
└── NXtomoproc.hdf5
```

**Review by Wiebke — APPROVED** [review]
Ingesting worked wonderfully on my end.
I ingested both tiff-sequences and an h5 files with this method. 

Some more notes on how I tested below:
Added a new ingesting service to the compose file from [mlex_tomo_framework](https://github.com/mlexchange/mlex_tomo_framework/), stared a shell in that service, and ran the `src/ingest.py` script.

Addition in docker-compose.yaml:
```yaml
  tiled_ingest:
    build:
      context: "../mlex_tiled_ingest/"
    command: sleep 99999
    environment:
      - TILED_DB_USER=${TILED_DB_USER}
      - TILED_DB_PW=${TILED_DB_PW}
      - TILED_DB_NAME=${TILED_DB_NAME}
    depends_on:
      - tiled_db
    volumes:
      - ./tiled/deploy:/deploy
      - ./data/tiled_storage:/tiled_storage
    networks:
      mle_net:
```

Exec into that service:
`podman-compose exec tiled_ingest bash`

Running the script:
`python src/ingest.py`

I did not test the RabbitMQ aspects, but noticed that now prefix path is given there. Our message may need to de-compose the received file_path?

File: src/ingest.py (hunk 2 above)
Comment (dylanmcreynolds) [reply src/ingest.py:87]: Yikes. Yeah, that `_unnamed` key was added not that [long ago](https://github.com/bluesky/tiled/pull/592). It's not great, but we couldn't come up with better.

Can we just fix the test data?

**Pull request merged**